diff --git a/src/libslic3r/ExPolygonCollection.cpp b/src/libslic3r/ExPolygonCollection.cpp
index a0de8f6de6..01027d77a2 100644
--- a/src/libslic3r/ExPolygonCollection.cpp
+++ b/src/libslic3r/ExPolygonCollection.cpp
@@ -77,7 +77,7 @@ bool
 ExPolygonCollection::contains_b(const Point &point) const
 {
     for (ExPolygons::const_iterator it = this->expolygons.begin(); it != this->expolygons.end(); ++it) {
-        if (it->contains_b(point)) return true;
+        if (it->contains(point)) return true;
     }
     return false;
 }
