diff --git a/src/libslic3r/Format/STEP.cpp b/src/libslic3r/Format/STEP.cpp
--- a/src/libslic3r/Format/STEP.cpp
+++ b/src/libslic3r/Format/STEP.cpp
@@ -542,8 +542,17 @@ Step::Step_Status Step::mesh(Model* model,

         std::vector<stl_file> stl;
         stl.resize(namedSolids.size());
+#ifdef __EMSCRIPTEN__
+        for (size_t i = 0; i < namedSolids.size(); i++) {
+#else
         tbb::parallel_for(tbb::blocked_range<size_t>(0, namedSolids.size()), [&](const tbb::blocked_range<size_t>& range) {
             for (size_t i = range.begin(); i < range.end(); i++) {
+#endif
+#ifdef __EMSCRIPTEN__
+                constexpr bool parallel_mesh = false;
+#else
+                constexpr bool parallel_mesh = true;
+#endif
-                BRepMesh_IncrementalMesh mesh(namedSolids[i].solid, linear_deflection, false, angle_deflection, true);
+                BRepMesh_IncrementalMesh mesh(namedSolids[i].solid, linear_deflection, false, angle_deflection, parallel_mesh);
                 // BBS: calculate total number of the nodes and triangles
                 int aNbNodes = 0;
@@ -616,7 +620,9 @@ Step::Step_Status Step::mesh(Model* model,
                 }
                 meshed_solid_num.fetch_add(1, std::memory_order_relaxed);
             }
+#ifndef __EMSCRIPTEN__
         });
+#endif


         for (size_t i = 0; i < stl.size(); i++) {
